### PR TITLE
Allow more options

### LINKED ISSUE #20: Add support for combination of metrics
Keda supports a formula based evaluation of metrics (https://keda.sh/docs/2.12/concepts/scaling-deployments).

### LINKED ISSUE #32: Support all metric types
- Depending on the metric there are different metric types to be supported: AverageValue, Value, Utilization.
- It holds:
> All scalers, except CPU and Memory, support metric types AverageValue and Value while CPU and Memory scalers both support AverageValue and Utilization.

### LINKED ISSUE #33: Support for multiple Prometheus Triggers
-  For example allow to scale on both cpu utilization and number of requests

### PR DESCRIPTION
**Pull request by skonto**
# Changes

- Add support for extra Prometheus triggers and scaled modifiers
- Allows to pass a scaled object directly via annotations in autocreate (managed mode).
- Adds documentation for the configuration options, annotations etc
- Minor clean up

Fixes #20 
Fixes #32
Fixes #33

### CODE REVIEW

File: CONCEPTS-API.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+# Introduction
+
+The goal of the Autoscaler KEDA extension is to keep the existing UX provided by the current Knative Serving and HPA integration.
+It builds on top of that to allow more fine-grained configurations and replaces the current HPA implementation with a more flexible and powerful one.
+The point of integration between KEDA and this extension is the existence of an HPA resource.
+If an HPA resource is available for a Knative Service, then the extension can reconcile the corresponding SKS in order networking setup to follow up.
```
Comment (ReToCode) [inline CONCEPTS-API.md:6]: maybe explain or link SKS.

File: pkg/reconciler/autoscaling/hpa/resources/keda.go [hunk 2]
```diff
@@ -184,3 +236,68 @@ func getDefaultPrometheusTrigger(annotations map[string]string, address string,
...
+	v, ok := annotations[KedaAutoscaleAnnotationMetricType]
+	if ok {
+		if v != string(autoscalingv2.AverageValueMetricType) && v != string(autoscalingv2.UtilizationMetricType) && v != string(autoscalingv2.ValueMetricType) {
+			return nil, fmt.Errorf("invalid metric type: %s", v)
+		}
+	}
+	switch metric {
+	case autoscaling.CPU:
+		if ok {
+			if v == string(autoscalingv2.ValueMetricType) {
```
Comment (ReToCode) [inline pkg/reconciler/autoscaling/hpa/resources/keda.go:261]: this seems to be repeated/duplicated with the above (l 254). Could the two checks be combined in the switch?

**Review by ReToCode — COMMENTED** [review]
See distinct comments.

File: CONCEPTS-API.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        containers:
+        - resources:
+            requests:
+              cpu: "100m"
+...
+```
+
+MinScale and maxScale define the minimum and maximum allowed replicas, they are optional and if not specified the extension will use the default values of 1 and infinite respectively.
+
+**Important** : At this point scale from zero is not supported, thus triggers should be always active.
```
[the remark this answers is not in the document]
Comment (skonto) [reply CONCEPTS-API.md:50]: https://keda.sh/docs/2.14/concepts/scaling-deployments

> ⚠️ NOTE: If the minimum replicas is >= 1, the scaler is always active and the activation value will be ignored.

File: CONCEPTS-API.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
+# Introduction
+
+The goal of the Autoscaler KEDA extension is to keep the existing UX provided by the current Knative Serving and HPA integration.
+It builds on top of that to allow more fine-grained configurations and replaces the current HPA implementation with a more flexible and powerful one.
+The point of integration between KEDA and this extension is the existence of an HPA resource.
+If an HPA resource is available for a Knative Service, then the extension can reconcile the corresponding SKS in order networking setup to follow up.
+
+In the current implementation at the core Serving, this HPA resource is created by the Serving controller based on the Pod Autoscaler (PA) resource for the current revision.
```
[the remark this answers is not in the document]
Comment (skonto) [reply CONCEPTS-API.md:8]: We don't create it we replace that component, upstream the hpa controller is a separate manifest from the core.

File: CONCEPTS-API.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ReToCode) [reply CONCEPTS-API.md:50]: I mean the part about triggers. You mean the prometheus triggers are always active as we will always be scaled to at least 1? IMHO, the latter part of the sentence can probably just be deleted. 

> At this point scale from zero is not supported

is what you wanted to say, right?

File: CONCEPTS-API.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ReToCode) [reply CONCEPTS-API.md:8]: Ah ok. Downstream to be figured out :D

File: CONCEPTS-API.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (skonto) [reply CONCEPTS-API.md:50]: Yes.

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**